### PR TITLE
adding functionality to increase only the current article

### PR DESCRIPTION
**Pull request by jm-w-a**
Adding functionality to increase only the current article

### CODE REVIEW

**Review by robert112233 — CHANGES_REQUESTED** [review]
Great work Joshua, the basic voting functionality is there. I think there are some features we could include on the frontend to provide a better experience:

- Allowing users to undo votes (if I click the button again maybe it undos the previous vote)
- Preventing multiple votes (currently I can just spam the votes button, making it obsolete) 
- Providing an error message if the vote didn't go through (if I vote offline it optimistically renders as normal) 

Please implement these changes and submit another PR for review :)

File: api.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+    const output = res.data.comments;
+    return output;
+  });
+};
+export const patchArticleVotes = (article_id, updatedValue) => {
+  return ncNewsApi
+    .patch(`/articles/${article_id}`, updatedValue)
+    .then((res) => {
+        const output = res.data;
+        console.log(output)
```
Comment (HeatherEver) [inline api.js:30]: please remove console logs :)

File: src/components/Article.jsx [hunk 2]
```diff
@@ -1,37 +1,72 @@
...
   }, [article_id]);
 
+  const handelVoteOnClick = () => {
+    setIsVoteError(false)
+    if(hasVoted){
+      setArticle((currArticle)=>{
+        return {...currArticle, votes: --currArticle.votes}
+      })
+      patchArticleVotes(article_id, {inc_votes: -1}).catch(()=>{
+        console.log('test')
```
Comment (HeatherEver) [inline src/components/Article.jsx:29]: and this one

File: src/components/Article.jsx [hunk 3]
```diff
@@ -1,37 +1,72 @@
...
+          return {...currArticle, votes: ++currArticle.votes}
+        })
+      })
+      setHasVoted(false)
+    }else{
+      setArticle((currArticle)=>{
+        return {...currArticle, votes: ++currArticle.votes}
+      })
+      patchArticleVotes(article_id, {inc_votes: 1}).catch(()=>{
+        console.log('test')
```
Comment (HeatherEver) [inline src/components/Article.jsx:42]: and this

**Review by HeatherEver — APPROVED** [review]
Good work implementing those changes ⭐ 
On a side note the Article component looks a little crowded now with vote logic - it would be nice if  you extracted that into it's on votes component to keep things neat, however this is not essential at this stage but something to consider going forward

**Pull request merged**